lib/project_types/script/layers/infrastructure/rust_project_creator.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+module Script
+  module Layers
+    module Infrastructure
+      class RustProjectCreator
+        include SmartProperties
+        property! :ctx, accepts: ShopifyCli::Context
+        property! :extension_point, accepts: Domain::ExtensionPoint
+        property! :script_name, accepts: String
+        property! :path_to_project, accepts: String
+
+        ORIGIN_BRANCH = 'main'
+
+        def setup_dependencies
+          git_init
+          setup_remote
+          setup_sparse_checkout
+        end
+
+        def bootstrap
+          pull
+          clean
+        end
+
+        private
+
+        def git_init
+          out, status = ctx.capture2e("git init")
+          raise Domain::Errors::ServiceFailureError, out unless status.success?
+        end
+
+        def setup_remote
+          repo = extension_point.sdks.rust.package
+          out, status = ctx.capture2e("git remote add -f origin #{repo}")
+          raise Domain::Errors::ServiceFailureError, out unless status.success?
+        end
+
+        def setup_sparse_checkout
+          type = extension_point.type
+          out, status = ctx.capture2e("git config core.sparsecheckout true")
+          raise Domain::Errors::ServiceFailureError, out unless status.success?
+          out, status = ctx.capture2e("echo #{type}/ >> .git/info/sparse-checkout")
```

Comment:
I need to add here `#{type}/default` which is where the default script will live (at least for the PoC).